Build fails for me:

```
/build/TeXmacs-1.99.13/src/Plugins/Qt/QTMPipeLink.cpp: In member function 'bool QTMPipeLink::launchCmd()':
/build/TeXmacs-1.99.13/src/Plugins/Qt/QTMPipeLink.cpp:52:39: warning: 'void QProcess::start(const QString&, QIODevice::OpenMode)' is deprecated: Use QProcess::start(const QString &program, const QStringList &arguments,OpenMode mode = ReadWrite) instead [-Wdeprecated-declarations]
   52 |   QProcess::start(utf8_to_qstring(cmd));
      |                                       ^
In file included from /nix/store/njywy8s62w93ymr822sc9cl03nv72q7r-qtbase-5.15.0-dev/include/QtCore/QProcess:1,
                 from /build/TeXmacs-1.99.13/src/Plugins/Qt/QTMPipeLink.hpp:17,
                 from /build/TeXmacs-1.99.13/src/Plugins/Qt/QTMPipeLink.cpp:15:
/nix/store/njywy8s62w93ymr822sc9cl03nv72q7r-qtbase-5.15.0-dev/include/QtCore/qprocess.h:168:10: note: declared here
  168 |     void start(const QString &command, OpenMode mode = ReadWrite);
      |          ^~~~~
[ 96%] Building CXX object src/CMakeFiles/texmacs_body.dir/Plugins/Qt/QTMWindow.cpp.o
[ 96%] Building CXX object src/CMakeFiles/texmacs_body.dir/Plugins/Qt/qt_chooser_widget.cpp.o
[ 96%] Building CXX object src/CMakeFiles/texmacs_body.dir/Plugins/Qt/qt_color_picker_widget.cpp.o
[ 96%] Building CXX object src/CMakeFiles/texmacs_body.dir/Plugins/Qt/qt_dialogues.cpp.o
/build/TeXmacs-1.99.13/src/Plugins/Qt/QTMStyle.cpp: In function 'void qtmDrawRoundedRect(QPainter*, const QRectF&, qreal, qreal, Qt::SizeMode)':
/build/TeXmacs-1.99.13/src/Plugins/Qt/QTMStyle.cpp:194:16: error: aggregate 'QPainterPath path' has incomplete type and cannot be defined
  194 |   QPainterPath path;
      |                ^~~~
make[2]: *** [src/CMakeFiles/texmacs_body.dir/build.make:6946: src/CMakeFiles/texmacs_body.dir/Plugins/Qt/QTMStyle.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/build/TeXmacs-1.99.13/src/Plugins/Qt/QTMWidget.cpp: In member function 'virtual void QTMWidget::paintEvent(QPaintEvent*)':
/build/TeXmacs-1.99.13/src/Plugins/Qt/QTMWidget.cpp:240:48: warning: 'QVector<QRect> QRegion::rects() const' is deprecated: Use begin()/end() instead [-Wdeprecated-declarations]
  240 |   QVector<QRect> rects = event->region().rects();
      |                                                ^
In file included from /nix/store/njywy8s62w93ymr822sc9cl03nv72q7r-qtbase-5.15.0-dev/include/QtGui/qmatrix.h:45,
                 from /nix/store/njywy8s62w93ymr822sc9cl03nv72q7r-qtbase-5.15.0-dev/include/QtGui/qbrush.h:49,
                 from /nix/store/njywy8s62w93ymr822sc9cl03nv72q7r-qtbase-5.15.0-dev/include/QtGui/qpalette.h:46,
                 from /nix/store/njywy8s62w93ymr822sc9cl03nv72q7r-qtbase-5.15.0-dev/include/QtWidgets/qwidget.h:48,
                 from /nix/store/njywy8s62w93ymr822sc9cl03nv72q7r-qtbase-5.15.0-dev/include/QtWidgets/qframe.h:44,
                 from /nix/store/njywy8s62w93ymr822sc9cl03nv72q7r-qtbase-5.15.0-dev/include/QtWidgets/qabstractscrollarea.h:44,
                 from /nix/store/njywy8s62w93ymr822sc9cl03nv72q7r-qtbase-5.15.0-dev/include/QtWidgets/QAbstractScrollArea:1,
                 from /build/TeXmacs-1.99.13/src/Plugins/Qt/QTMScrollView.hpp:15,
                 from /build/TeXmacs-1.99.13/src/Plugins/Qt/QTMWidget.hpp:16,
                 from /build/TeXmacs-1.99.13/src/Plugins/Qt/QTMWidget.cpp:12:
/nix/store/njywy8s62w93ymr822sc9cl03nv72q7r-qtbase-5.15.0-dev/include/QtGui/qregion.h:125:20: note: declared here
  125 |     QVector<QRect> rects() const;
      |                    ^~~~~
make[1]: *** [CMakeFiles/Makefile2:317: src/CMakeFiles/texmacs_body.dir/all] Error 2
make: *** [Makefile:149: all] Error 2
```

--- a/pkgs/applications/editors/texmacs/default.nix
+++ b/pkgs/applications/editors/texmacs/default.nix
@@ -1,5 +1,5 @@
-{ stdenv, callPackage,
-  fetchurl, guile_1_8, qt4, xmodmap, which, makeWrapper, freetype,
+{ lib, mkDerivation, callPackage, fetchurl, fetchpatch,
+  guile_1_8, qtbase, xmodmap, which, freetype,
   libjpeg,
   sqlite,
   tex ? null,
@@ -16,23 +16,30 @@
 
 let
   pname = "TeXmacs";
-  version = "1.99.11";
+  version = "1.99.13";
   common = callPackage ./common.nix {
     inherit tex extraFonts chineseFonts japaneseFonts koreanFonts;
   };
 in
-stdenv.mkDerivation {
+mkDerivation {
   name = "${pname}-${version}";
 
   src = fetchurl {
     url = "https://www.texmacs.org/Download/ftp/tmftp/source/TeXmacs-${version}-src.tar.gz";
-    sha256 = "12bp0f34izzqimz49lfpgf4lyz3h45s9xbmk8v6zsawdjki76alg";
+    sha256 = "Aq0cS47QqmFQHelxRjANeJlgXCXagnYRykpAq7wHqbQ=";
   };
 
-  cmakeFlags = [
-    # Texmacs' cmake build as of writing defaults to Qt5,
-    # but we haven't updated to that yet.
-    "-DTEXMACS_GUI=Qt4"
+  patches = [
+    # Minor patch for Qt 5.15 support, should be included in next release.
+    (fetchpatch {
+      url = "https://github.com/texmacs/texmacs/commit/3cf56af92326b74538f5e943928199ba6e963d0b.patch";
+      sha256 = "+OBQmnKgvQZZkLx6ea773Dwq0o7L92Sex/kcVUhmg6Q=";
+    })
+    # Fix returned version, lets hope they remember to bump the version next release.
+    (fetchpatch {
+      url = "https://github.com/texmacs/texmacs/commit/da5b67005d2fc31bb32ea1ead882c26af12d8cbb.patch";
+      sha256 = "czMgdraQErrdvN83jY76P673L52BpQkDwntmKvF0Ykg=";
+    })
   ];
 
   enableParallelBuilding = true;
@@ -40,8 +47,7 @@ stdenv.mkDerivation {
   nativeBuildInputs = [ cmake pkgconfig ];
   buildInputs = [
     guile_1_8
-    qt4
-    makeWrapper
+    qtbase
     ghostscriptX
     freetype
     libjpeg
@@ -51,18 +57,26 @@ stdenv.mkDerivation {
   ];
   NIX_LDFLAGS = "-lz";
 
-  postInstall = "wrapProgram $out/bin/texmacs --suffix PATH : " +
-        (if ghostscriptX == null then "" else "${ghostscriptX}/bin:") +
-        (if aspell == null then "" else "${aspell}/bin:") +
-        (if tex == null then "" else "${tex}/bin:") +
-        (if git == null then "" else "${git}/bin:") +
-        (if python3 == null then "" else "${python3}/bin:") +
-        "${xmodmap}/bin:${which}/bin";
+  qtWrapperArgs = [
+    "--suffix" "PATH" ":" (lib.makeBinPath [
+      xmodmap
+      which
+      ghostscriptX
+      aspell
+      tex
+      git
+      python3
+    ])
+  ];
+
+  postFixup = ''
+    wrapQtApp $out/bin/texmacs
+  '';
 
   inherit (common) postPatch;
 
   meta = common.meta // {
-    maintainers = [ stdenv.lib.maintainers.roconnor ];
-    platforms = stdenv.lib.platforms.gnu ++ stdenv.lib.platforms.linux;  # arbitrary choice
+    maintainers = [ lib.maintainers.roconnor ];
+    platforms = lib.platforms.gnu ++ lib.platforms.linux;  # arbitrary choice
   };
 }

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7354,7 +7354,7 @@ in
       inherit (darwin.apple_sdk.frameworks) CoreFoundation Cocoa;
       tex = texlive.combined.scheme-small;
       extraFonts = true;
-    } else callPackage ../applications/editors/texmacs {
+    } else libsForQt5.callPackage ../applications/editors/texmacs {
       tex = texlive.combined.scheme-small;
       extraFonts = true;
     };

